src/figuresClasses.ts

```diff
@@ -1,19 +1,120 @@
-export interface Figure {
+function roundNumber(number: number, digitsAfterComma: number): number {
+  return Math.trunc(number * (10 ** digitsAfterComma))
+    / (10 ** digitsAfterComma);
+}
 
+enum Shape{
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color{
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  getType(): Shape;
+  getColor(): Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  private shape: Shape;
+
+  constructor(
+    private color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if ((a + b <= c)
+     || (b + c <= a)
+     || (a + c <= b)
+    ) {
+      throw new Error('sides 1, 2 and 3 can\'t form a triangle');
+    }
+
+    this.shape = Shape.Triangle;
```

Comment:
Each class has own shape property, so you can write your enum straight to it
for example
`private shape: Shape = Shape.Triangle`
so you can delete this line here and in the next classes